en/lessons/basics/modules.md

```diff
@@ -1,316 +1,196 @@
+a module, and what modules can be composed into. The elements that can be
+exported from a module are called entities. In Haskell those are:
 
-Let's look at a basic example:
+- Types and type synonyms
+- Values: constants and functions
+- Typeclasses
 
-```elixir
-defmodule Example do
-  def greeting(name) do
-    "Hello #{name}."
-  end
-end
+Modules are really just collections of these entities that are exported.
+Modules can be combined through a graph of imports and exports, this is
+sometimes called a dependency graph. These dependency graphs can be packaged
+together as a component. The two most common components are libraries and
+executables. A library allows you to export your modules for re-use in other
+codebases, and an executable consists of an entrypoint (a Main module) to the
+dependency graph that can be run as a program. A good example of a library is
+`base` and `prelude` these libraries ship with GHC, `lens` is a good example of
+a third party library that requires a package manager to import. A good example
+of executables are `xmonad` and `pandoc`. Haskell's most popular package
+managers also have executables that allow them to be run from the command line;
+`cabal-install` and `stack`.
 
-iex> Example.greeting "Sean"
-"Hello Sean."
-```
-
-It is possible to nest modules in Elixir, allowing you to further namespace your functionality:
-
-```elixir
-defmodule Example.Greetings do
-  def morning(name) do
-    "Good morning #{name}."
-  end
-
-  def evening(name) do
-    "Good night #{name}."
-  end
-end
-
-iex> Example.Greetings.morning "Sean"
-"Good morning Sean."
-```
-
-### Module Attributes
-
-Module attributes are most commonly used as constants in Elixir.
-Let's look at a simple example:
-
-```elixir
-defmodule Example do
-  @greeting "Hello"
-
-  def greeting(name) do
-    ~s(#{@greeting} #{name}.)
-  end
-end
-```
-
-It is important to note there are reserved attributes in Elixir.
-The three most common are:
+To reiterate, entities are our basic language-level building blocks. Modules
+are groupings of entities with some or all exported. Components are graphs of
+modules that can be made available in other codebases or executed via an
+entrypoint module.
 
-- `moduledoc` — Documents the current module.
-- `doc` — Documentation for functions and macros.
-- `behaviour` — Use an OTP or user-defined behaviour.
+## Module Syntax
 
-## Structs
+Module syntax is driven by 2 keywords `module` and `where`. The `module` keyword
+precedes the modules name, which must match the filename, and optionally a
+collection of entities to export. The `where` keyword ends the module
+declaration and indicates the beginning of the module.
 
-Structs are special maps with a defined set of keys and default values.
-A struct must be defined within a module, which it takes its name from.
-It is common for a struct to be the only thing defined within a module.
+Module declaration exposing all entities.
```

Comment:
It may be unusual for the reader to be introduced to a snippet of code like this. A word like "here" may be useful here.